### PR TITLE
Dragging - Add possiblity to rotate a carried object

### PR DESCRIPTION
**Pull request by bux**
**When merged this pull request will:**
- title
- use CBA_fnc_addDisplayHandler
- fix wrong number of param in showMouseHint comment

### CODE REVIEW

**Comment by bux** [comment]
Example: https://streamable.com/qfaxkv

File: addons/dragging/functions/fnc_onKeyDown.sqf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Public: No
+ */
+
+// EH is always assigned. Exit and don't overwrite input if not carrying
+if !(ace_player getVariable [QGVAR(isCarrying), false]) exitWith {false};
+
+params ["", "_key", "_shift", "_ctrl", "_alt"];
+
+GVAR(ctrlHeld) = _ctrl;
```
Comment (PabstMirror) [inline addons/dragging/functions/fnc_onKeyDown.sqf:24]: just use `cba_events_control`

Comment (PabstMirror) [reply addons/dragging/functions/fnc_onKeyDown.sqf:24]: e.g. https://github.com/acemod/ACE3/blob/master/addons/explosives/functions/fnc_openTimerUI.sqf#L58

Comment (bux) [reply addons/dragging/functions/fnc_onKeyDown.sqf:24]: Nice! That's so much better :)

File: addons/dragging/stringtable.xml [hunk 2]
```diff
@@ -86,21 +86,9 @@
...
-            <Italian>Alza/Abbassa</Italian>
-            <Spanish>Subir/Bajar</Spanish>
-            <French>Lever/Baisser</French>
-            <Japanese>上げる/下げる</Japanese>
-            <Korean>높이기/낮추기</Korean>
-            <Chinese>提高/下降</Chinese>
-            <Chinesesimp>提高/下降</Chinesesimp>
-            <Turkish>Yukarı/Aşağı</Turkish>
+        <Key ID="STR_ACE_Dragging_RaiseLowerRotate">
+            <English>Raise/Lower | (Ctrl + Scroll) Rotate</English>
```
Comment (commy2) [inline addons/dragging/stringtable.xml:90]:
```suggestion
            <English>Raise/Lower | (Ctrl + Scroll Wheel) Rotate</English>
```

File: addons/dragging/stringtable.xml [hunk 3]
```diff
@@ -86,21 +86,9 @@
...
+            <German>Heben/Senken | (Strg + Scrollen) Drehen</German>
```
Comment (commy2) [inline addons/dragging/stringtable.xml:91]:
```suggestion
            <German>Heben/Senken | (Strg + Scrollrad) Drehen</German>
```

Scroll is not a key. It shoul

Comment (bux) [reply addons/dragging/stringtable.xml:91]: Same as here:
https://github.com/acemod/ACE3/blob/master/addons/advanced_throwing/stringtable.xml#L247

File: addons/dragging/functions/fnc_handleScrollWheel.sqf [hunk 4]
```diff
@@ -17,35 +17,47 @@
...
 
-private _unit = ACE_player;
-
 // EH is always assigned. Exit and don't overwrite input if not carrying
-if !(_unit getVariable [QGVAR(isCarrying), false]) exitWith {false};
+if !(ace_player getVariable [QGVAR(isCarrying), false]) exitWith {false};
 
-// move carried item 15 cm per scroll interval
-_scrollAmount = _scrollAmount * 0.15;
+private _unit = ace_player;
```
Comment (commy2) [inline addons/dragging/functions/fnc_handleScrollWheel.sqf:23]: ACE_player.

Why move it down here though?

Comment (bux) [reply addons/dragging/functions/fnc_handleScrollWheel.sqf:23]: I don't want to create a variable each time the player uses it's scrollwheel without carrying an item.

File: addons/dragging/stringtable.xml (hunk 3 above)
Comment (commy2) [reply addons/dragging/stringtable.xml:91]: Then change it there as well.

File: addons/dragging/functions/fnc_handleScrollWheel.sqf (hunk 4 above)
Comment (commy2) [reply addons/dragging/functions/fnc_handleScrollWheel.sqf:23]: You're not going to save a single frame by this in 1000 hours of playing A3 while continiuously turning the mouse wheel.
I don't even think your mouse will survive before you saved one second.

File: addons/dragging/stringtable.xml (hunk 3 above)
Comment (bux) [reply addons/dragging/stringtable.xml:91]: No. Can be done in a separate PR.

Comment (jonpas) [reply addons/dragging/stringtable.xml:91]: No, do not change it. Scroll is scroll wheel rolling there is also a Middle Mouse Button as key which is also used in Advanced Throwing.

**Comment by PabstMirror** [comment]
Would this be any better, it matches what we use on acex_fortify:

**Comment by bux** [comment]
I used the same as in adv. throwing. Might be for a future improvement where we change this to use a small key cap icon <kbd>Ctrl</kbd>

File: addons/dragging/functions/fnc_carryObject.sqf [hunk 5]
```diff
@@ -63,7 +63,7 @@ _unit setVariable [QGVAR(ReleaseActionID), [
 if (_target isKindOf "CAManBase") then {
     [localize LSTRING(Drop), "", ""] call EFUNC(interaction,showMouseHint);
 } else {
-    [localize LSTRING(Drop), "", localize LSTRING(LowerRaise)] call EFUNC(interaction,showMouseHint);
+    [localize LSTRING(Drop), "", localize LSTRING(RaiseLowerRotate)] call EFUNC(interaction,showMouseHint);
```
Comment (PabstMirror) [inline addons/dragging/functions/fnc_carryObject.sqf:66]:
```suggestion
    [localize LSTRING(Drop), "", LLSTRING(LowerRaise), [["control", LLSTRING(rotate)]]] call EFUNC(interaction,showMouseHint);
```
The advantage of doing this now is we don't have to drop our existing translations

Comment (bux) [reply addons/dragging/functions/fnc_carryObject.sqf:66]: It's hideous, though

**Pull request merged**